src/layer/marker/Icon.Default.js

```diff
@@ -1,45 +1,51 @@
 /*
- * L.Icon.Default is the blue marker icon used by default in Leaflet.
+ * @miniclass Icon.Default (Icon)
+ * @aka L.Icon.Default
+ * @section
+ *
+ * A trivial subclass of `Icon`, represents the icon to use in `Marker`s when
+ * no icon is specified. Points to the blue marker image distributed with Leaflet
+ * releases.
+ *
+ * In order to change the default icon, just change the properties of `L.Icon.Default.prototype.options`
+ * (which is a set of `Icon options`).
  */
 
 L.Icon.Default = L.Icon.extend({
 
 	options: {
+		iconUrl:       'marker-icon.png',
+		iconRetinaUrl: 'marker-icon-2x.png',
+		shadowUrl:     'marker-shadow.png',
 		iconSize:    [25, 41],
 		iconAnchor:  [12, 41],
 		popupAnchor: [1, -34],
 		shadowSize:  [41, 41]
 	},
 
 	_getIconUrl: function (name) {
-		var key = name + 'Url';
 
-		if (this.options[key]) {
-			return this.options[key];
-		}
-
-		var path = L.Icon.Default.imagePath;
+		// @option imagePath: String
+		// `L.Icon.Default` will try to auto-detect the absolute location of the
+		// blue icon images. If you are placing these images in a non-standard
+		// way, set this option to point to the right absolute path.
+		if (!('imagePath' in this.options)) {
 
-		if (!path) {
-			throw new Error('Couldn\'t autodetect L.Icon.Default.imagePath, set it manually.');
-		}
+			if (L.Icon.Default.imagePath) {	// Deprecated, backwards-compatibility only
+				this.options.imagePath = L.Icon.Default.imagePath;
```

Comment:
There's no need to reset the original `imagePath` property in options. I think this should be just a variable, used throughout the function. Otherwise I think this is ready to merge!